core/templates/services/context.service.spec.ts

```diff
@@ -246,12 +260,47 @@ describe('Context service', () => {
       });
   });
 
+  describe('behavior in different pages', () => {
+    beforeEach(() => {
+      ecs = TestBed.get(ContextService);
+      urlService = TestBed.get(UrlService);
+    });
+
+    it('should correctly retrieve the page context', () => {
+      spyOn(urlService, 'getPathname').and.returnValue('/question_editor/123');
+      expect(ecs.getPageContext()).toBe('question_editor');
+    });
+
+    it('should correctly retrieve the page context', () => {
+      spyOn(urlService, 'getPathname').and.returnValue('/session/123');
+      expect(ecs.getPageContext()).toBe('question_player');
+    });
+
+    it('should correctly retrieve the page context', () => {
+      spyOn(urlService, 'getPathname').and.returnValue(
+        '/collection_editor/123');
+      expect(ecs.getPageContext()).toBe('collection_editor');
+    });
+
+    it('should correctly retrieve the page context', () => {
+      spyOn(urlService, 'getPathname').and.returnValue(
+        '/topics-and-skills-dashboard/123');
+      expect(ecs.getPageContext()).toBe('topics_and_skills_dashboard');
+    });
+
+    it('should correctly retrieve the page context', () => {
+      spyOn(urlService, 'getPathname').and.returnValue(
+        '/contributor-dashboard/123');
+      expect(ecs.getPageContext()).toBe('contributor_dashboard');
+    });
+  });
+
   describe('behavior in other pages', () => {
     beforeEach(() => {
       ecs = TestBed.get(ContextService);
       urlService = TestBed.get(UrlService);
       spyOn(urlService, 'getPathname').and.returnValue('/about');
-      spyOn(urlService, 'getHash').and.returnValue('');
+      spyOn(urlService, 'getHash').and.returnValue('#/preview');
```

Comment:
Why change this?

Reply:
So `getEditorTabContext()` returns 'preview' which is tested below in spec.

Reply:
Update: removed `spyOn(urlService, 'getHash').and.returnValue('');` from beforeEach() and using it in spec below to include pre-check